### PR TITLE
Scottx611x/docker puppet install

### PR DESCRIPTION
**Pull request by scottx611x**

### CODE REVIEW

File: deployment/Puppetfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 forge "https://forgeapi.puppetlabs.com"
 
+mod 'garethr-docker', '5.3.0'
```
Comment (hackdna) [inline deployment/Puppetfile]: Why not use the [puppetlabs/docker_platform](https://forge.puppet.com/puppetlabs/docker_platform)?

File: deployment/refinery-modules/refinery/manifests/docker_.pp [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class refinery::docker_ {
```
Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/docker_.pp]: Why not `refinery::docker`?

File: deployment/refinery-modules/refinery/manifests/init.pp [hunk 3]
```diff
@@ -229,6 +226,7 @@
     Class["ui"],
     Class["solr"],
     Class["neo4j"],
+    Class["docker"],
```
Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/init.pp]: Why does supervisor require docker?

File: deployment/refinery-modules/refinery/manifests/selenium.pp [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class refinery::selenium {
   $geckodriver_version = 'v0.15.0'
-  $filename = "geckodriver-${geckodriver_version}-linux32.tar.gz"
+  $filename = "geckodriver-${geckodriver_version}-linux64.tar.gz"
```
Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/selenium.pp]: Why is this necessary?

File: deployment/refinery-modules/refinery/manifests/docker_.pp [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+  class { 'docker':
```
Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/docker_.pp]: This should be `class { '::docker':` [to indicate global namespace explicitly](http://serverfault.com/questions/421023/why-does-my-namespace-and-class-name-conflict)

File: deployment/Puppetfile (hunk 1 above)
Comment (scottx611x) [inline deployment/Puppetfile]: Not sure, it looks exactly the same as: `garethr-docker`
Actually it depends on it: https://forge.puppet.com/puppetlabs/docker_platform/dependencies

File: deployment/refinery-modules/refinery/manifests/docker_.pp (hunk 2 above)
Comment (scottx611x) [inline deployment/refinery-modules/refinery/manifests/docker_.pp]: Because there is already a class`docker` and Puppet complains about a naming conflict

File: deployment/refinery-modules/refinery/manifests/docker_.pp (hunk 5 above)
Comment (scottx611x) [inline deployment/refinery-modules/refinery/manifests/docker_.pp]: Okay, should this be changed throughout our manifests? I was just following the syntax that we had used throughout

File: deployment/refinery-modules/refinery/manifests/selenium.pp (hunk 4 above)
Comment (scottx611x) [inline deployment/refinery-modules/refinery/manifests/selenium.pp]: Because Geckodriver is not providing 32 bit builds in its newer releases anymore

File: deployment/refinery-modules/refinery/manifests/init.pp (hunk 3 above)
Comment (scottx611x) [inline deployment/refinery-modules/refinery/manifests/init.pp]: I don't think supervisor explicitly needs docker, but i don't think we want to start up everything since our app will depend on docker working.

File: deployment/refinery-modules/refinery/manifests/docker_.pp (hunk 5 above)
Comment (scottx611x) [inline deployment/refinery-modules/refinery/manifests/docker_.pp]: Also this answer is from 2012, is it still relevant?

File: deployment/Puppetfile (hunk 1 above)
Comment (hackdna) [inline deployment/Puppetfile]: The benefit of the puppetlabs modules is that they are provided by the Puppet team. Also, garethr-docker was last updated about a year ago and it supports only up to Ubuntu 14.04. The puppetlabs version was last updated a week ago and supports Ubuntu 16.04. Basically, it looks like it is probably going to replace garethr-docker.

File: deployment/refinery-modules/refinery/manifests/init.pp (hunk 3 above)
Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/init.pp]: I'd remove that since there is no explicit dependency and docker installation will still happen by the time the Refinery manifest is applied.

File: deployment/refinery-modules/refinery/manifests/docker_.pp (hunk 5 above)
Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/docker_.pp]: Yes, it's the reason you got the naming error when trying to use `refinery::docker` on the line above. Here's the source in Puppet docs (still relevant as of Puppet v3.5 and we are using v3.4.3): https://docs.puppet.com/puppet/3.5/lang_namespaces.html#workaround

Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/docker_.pp]: And yes, it should be changed in the rest of the manifests but in a separate branch.

File: deployment/refinery-modules/refinery/manifests/docker.pp [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class refinery::docker {
+  class { '::docker':
+    docker_users => [$app_user]
```
Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/docker.pp]: `$::app_user`

Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/docker.pp]: `$::app_user`
